library/Icingadb/Web/Controller.php

```diff
@@ -349,19 +360,82 @@ public function createViewModeSwitcher()
             $viewModeSwitcher->setDefaultViewMode($viewMode);
         }
 
-        $viewModeSwitcher->populate(['view' => $this->params->shift($viewModeSwitcher->getViewModeParam())]);
+        $viewModeSwitcher->populate([
+            'view' => $this->params->shift($viewModeSwitcher->getViewModeParam())
+        ]);
 
-        $viewModeSwitcher->on(ViewModeSwitcher::ON_SUCCESS, function (ViewModeSwitcher $viewModeSwitcher) use ($prefs) {
-            $viewMode = $viewModeSwitcher->getValue($viewModeSwitcher->getViewModeParam());
+        if (
+            $viewModeSwitcher->getViewMode() === 'minimal'
+            || Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal'
```

Comment:
Above, the call to populate already populates the request's view mode so that `getViewMode()` *can* return it.

So if it does, the second condition is redundant and won't ever be evaluated. (Because if `minimal` is indeed the request's view mode, it's also the switcher's current view mode.)

Yes, it currently does not return the request's view mode, because the view mode switcher isn't assembled yet. `getViewMode()` needs to call `ensureAssembled()` internally before the call to `getValue()`. This way it does return the request's view mode. This should play fine with other usages, as every other call to `getViewMode()` already does so when the view mode switcher has been assembled. (`handleRequest()` assembles)